src/region.rs

```diff
@@ -59,3 +61,28 @@ impl From<u8> for Region {
         Region::from(x as usize)
     }
 }
+
+impl FromStr for Region {
+    type Err = Error;
```

Comment:
Error is never returned make this infallible https://doc.rust-lang.org/std/convert/enum.Infallible.html